src/randomizer.py

```diff
@@ -1745,6 +1745,77 @@ def collect_binops(node: ca.BinaryOp) -> None:
     ast_util.insert_statement(ins_block, ins_index, new_assign)
 
 
+def perm_remove_ast(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Delete parts of the function that might be unnecessary (mistakes or unnecessary changes from an improved base.c)."""
+    cands = []
+
+    class Visitor(ca.NodeVisitor):
+        def visit_Cast(self, node: ca.Cast) -> None:
+            if region.contains_node(node):
+                cands.append(node)
+
+        # Replace (a & constant) with (a).
+        def visit_BinaryOp(self, node: ca.BinaryOp) -> None:
+            if region.contains_node(node) and node.op == "&":
+                if isinstance(node.left, ca.Constant) or isinstance(
+                    node.right, ca.Constant
+                ):
+                    cands.append(node)
+
+        # Remove if statements that don't have an else
+        def visit_If(self, node: ca.If) -> None:
+            if node.iffalse:
+                return
+
+            if region.contains_node(node):
+                cands.append(node)
+
+    Visitor().visit(fn.body)
+    ensure(cands)
+
+    cand = random.choice(cands)
+    if isinstance(cand, ca.Cast):
+        expr = cand.expr
+    elif isinstance(cand, ca.BinaryOp):
+        if isinstance(cand.left, ca.Constant) and isinstance(cand.right, ca.Constant):
+            expr = random.choice([cand.left, cand.right])
+        elif isinstance(cand.left, ca.Constant):
+            expr = cand.right
+        else:
+            expr = cand.left
+    elif isinstance(cand, ca.If):
+        expr = cand.iftrue
```

Comment:
Instead of isinstance checks here, we could have `cands` contain tuples of "(expr, expr to replace it with)" and move code into the visitor.

This would also simplify BinaryOp since it can add two entries into cands if both lhs and rhs are literals.